### ISSUE TITLE
Separate MonoGame.Net Repository

### ISSUE THREAD

**Issue by flibitijibibo** [mono/MonoGame#3055]
This is a follow-up to #2514, #438, and the following article by Simon Jackson:

http://darkgenesis.zenithmoon.com/monogame-modularisation/

FNA's repository has now been reorganized... (EDIT)

https://github.com/flibitijibibo/FNA

... so I'm now more interested in figuring out a dedicated repository for the now separate MonoGame.Framework.Net.dll project.

I've put all of the GamerServices/Net code from FNA into this repo, for convenience:

https://github.com/flibitijibibo/MonoGame.Net

As far as I remember there have been no actual code changes in these folders except to add stub methods. Any platform-specific files were based on the Linux files, since most of the desktop platforms were using the same implementation anyway (that is, all stub methods). The full log for both folders is here (the only changes made have been by me):

https://github.com/flibitijibibo/MonoGame/commits/monogame-sdl2/MonoGame.Framework/GamerServices
https://github.com/flibitijibibo/MonoGame/commits/monogame-sdl2/MonoGame.Framework/Net

The current setup MonoGame has for generating the projects would be the same as it is now, except I will also be providing an FNA configuration as well. The goal is to have everyone using the same networking code that functions across all XNA implementations (including Microsoft XNA4), both to reduce duplicated effort and simplify cross-platform networking (particularly if you plan to roll your own proprietary implementation of these namespaces; Steamworks versions based on the MonoGame GamerServices/Net code are a common example).

How the separate repository is integrated into the main MonoGame repository, and how the migration will play out, I will of course leave to the MonoGame team. Once the repo is ready (however it is made ready), I'll submit the FNA configuration ASAP.

**Comment by tgjones** [comment]
@tomspilman @KonajuGames can a decision be made on this? Presumably the defacto answer after a year and a half is "no", but it would be useful to clarify.

**Comment by tomspilman** [comment]
> Presumably the defacto answer after a year and a half is "no", but it would be useful to clarify.

Yeah...  I think it is a no.

 - Another repo is just more work for us managing MonoGame.
 - MonoGame.Framework.Net has lots of dependencies on the core of MonoGame.
 - The existing XNA network API doesn't map well to things like Stream, PSN, XBL, etc.
 - Our network implementation is not all that good.

Thoughts @dellis1972 @KonajuGames ?

**Comment by cra0zy** [comment]
If I can say something, I wouldn't like a separate repo for it, but it should get it's own folder.

**Comment by tomspilman** [comment]
>  I don't think it should get a separate repo, but it should get it's own folder.

That is probably a good idea...  just like the content pipeline is in its own top level folder.

**Comment by DDReaper** [comment]
I'd agree, moving it to it's own folder would be prudent to separate it out from the main project.
I recall when I was doing the Windows 10 presentation a few changes were committed to the .NET namespace as some of the guide features didn't work as expected with Windows 10.

**Comment by tomspilman** [comment]
Ok... that seems like a reasonable first step at least.

Want to see what @dellis1972 thinks as I think he was doing some work in this area before.

**Comment by dellis1972** [comment]
Moving to a new Folder is probably a good idea. 
`MonoGame.Framework.Net`
A separate repo would be a bad idea. I'm messing with the networking stuff on and off trying to drop the dependency on Lidgren. 

On a side note, the networking does still work on Linux,Mac,Windows,iOS and Android. I ported NetRumble the other week without any issues.

**Comment by tomspilman** [comment]
> On a side note, the networking does still work 

Yeah... we have to be specific in what we mean with that.  Firing packets off at each other works...   friends lists, match making, leaderboards, etc. does not.

**Comment by DDReaper** [comment]
Would love to see an Azure sample using the .NET lib, or better yet a Azure relacement using the API shim of the .NET project.
Even better to have an Xbox Live version but that would have to be private because of the ID NDA for Windows 10

**Comment by tomspilman** [comment]
Closing this in favor of https://github.com/mono/MonoGame/issues/4794.
